src/HeaderSetPlugin.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Http\Client\Plugin;
+
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Set headers to the request. Will override values if a header is already set
+ */
+class HeaderSetPlugin implements Plugin
+{
+
+    private $headers = [];
+
+    /**
+     * @param array $headers headers to set to the request
+     */
+    public function __construct(array $headers)
+    {
+        $this->headers = $headers;
+    }
+
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handleRequest(RequestInterface $request, callable $next, callable $first)
+    {
+        foreach ($this->headers as $header => $headerValue) {
+            $request = $request->withHeader($header, $headerValue);
```

Comment:
Yes it will replace, you can check here: http://www.php-fig.org/psr/psr-7/#headers-with-multiple-values